### PR TITLE
added boss monsters in the game

### PR DESCRIPTION
**Pull request by siddarthpai**

### CODE REVIEW

**Review by Himank101 — CHANGES_REQUESTED** [review]
Hey, thanks for working on this issue!
Please remove all the unnecessary files in your PR
make your changes only in the GUI2.py file

**Comment by siddarthpai** [comment]
sure im wokring on it

**Review by siddarthpai — COMMENTED** [review]
removed unnecessary files

File: GUI2.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 which_potion = 0  # variable that lets you select the potion that you want to take.
 
 def get_room():
-    room = ("monster", "shop", "treasure box", "monster", "shop")
+    room = ("monster", "shop", "treasure box", "bossmonster", "shop")
```
Comment (Himank101) [inline GUI2.py:23]: This is not feasible since we want a certain number of rooms to be crossed before entry is allowed in the boss room

**Review by Himank101 — CHANGES_REQUESTED** [review]
Try changing the way get_room() is called
Put the function in a for loop and called the get_bosmonsters() outside it

**Comment by Himank101** [comment]
@siddarthpai Kindly provide an update in under an hour.

File: GUI2.py [hunk 2]
```diff
@@ -754,6 +762,41 @@ def get_monster():
...
+    L_monster_Wel.pack()
+    bossmonsters = ("Demon Slayer","Big Tooth")
+    monster = random.choice(bossmonsters)
+    # print(monster)
+    opp_hp = 100
+    if monster == "Demon Slayer":
+        m = 1
+        # monster 1
+        # Attack in range of 0-10
+        L_m1_intro = Label(frame_monster_1, text="You have to face Goblin\n"
```
Comment (Himank101) [inline GUI2.py:782]: change text to the name of the monster in the text

File: GUI2.py [hunk 3]
```diff
@@ -754,6 +762,41 @@ def get_monster():
...
+                                                 "The match starts. You get the first chance\n")
+        L_m1_intro.pack()
+        fight_monster()
+        # opp_att = random.randint(0, 10)
+
+    if monster == "Big Tooth":
+        m = 2
+        # monster 2
+        # Attack in range of 10-20
+        L_m2_intro = Label(frame_monster_1, text="You have to face Werewolf\n"
```
Comment (Himank101) [inline GUI2.py:792]: Change name of the monster in the text

File: GUI2.py [hunk 4]
```diff
@@ -754,6 +762,41 @@ def get_monster():
...
+                                                 "The match starts. You get the first chance\n")
+        L_m2_intro.pack()
+        fight_monster()
+        # opp_att = random.randint(10, 20)
+
+
+
```
Comment (Himank101) [inline GUI2.py:799]: Call the quit function after get_bossmonster is executed or make a custom ending screen(preferable)

**Review by Himank101 — CHANGES_REQUESTED** [review]
fix a few tiny errors and make an ending screen. The rest of the implementation is good!

**Comment by siddarthpai** [comment]
working on it

File: GUI2.py [hunk 5]
```diff
@@ -754,6 +762,46 @@ def get_monster():
         # opp_att = random.randint(50, 60)
 
 
+def quit_screen():
+    quit_scr= Label(frame_monster_1, text="Game over")
+    quit_scr.pack()
```
Comment (Himank101) [inline GUI2.py:767]: Add a button that runs the quit function and it'll be perfect✨

**Review by Himank101 — COMMENTED** [review]
just add a button that executes the quit function onclick

**Comment by Himank101** [comment]
Congrats!! Now we have nemesis for our player!!

**Comment by Himank101** [comment]
!bounty 55

**Pull request merged**